src/include/error.h

```diff
@@ -67,14 +67,16 @@
 	int __ret;							\
 	if ((__ret = (a)) != 0 &&					\
 	    (__ret == WT_PANIC ||					\
-	    ret == 0 || ret == WT_DUPLICATE_KEY || ret == WT_NOTFOUND))	\
+	    ret == 0 || ret == WT_DUPLICATE_KEY ||			\
+	    ret == WT_NOTFOUND || ret == WT_RESTART))			\
```

Comment:
> I'd rather `WT_RESTART` was handled in the case where it's relevant. I'm concerned that having a handler in here will hide other errors that should be handled explicitly.

We may not be understanding each other: to me, the whole point of this change is to not hide other errors behind `WT_RESTART`.

I can't think of any case where `WT_RESTART` should override `ENOMEM` (or pretty much any other error I can imagine). Let me know if you can think of cases where we'd still want to proceed with operation retry after a different error occurred.